AsyncDisplayKit/ASViewController.mm

```diff
@@ -132,4 +142,70 @@ - (ASInterfaceState)interfaceState
   return _node.interfaceState;
 }
 
+#pragma mark - ASEnvironmentTraitCollection
+
+- (ASEnvironmentTraitCollection)displayTraitsForTraitCollection:(UITraitCollection *)traitCollection
```

Comment:
Ultimately we are passing around trait collections as a struct, not a class. It has to be a struct to live in ASEnvironment. The class is just there to be a nicer API to the end user (that probably doesn't care about the context).